src/components/EditEventDialog.tsx

```diff
@@ -0,0 +1,69 @@
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { useState } from "react";
+import EventForm from "./Form/EventForm";
+// import InputFieldWithLabel from "./Form/InputFieldWithLabel";
+import { IEditEventDialogProps } from "@/types/interface";
+
+const EditEventDialog: React.FC<IEditEventDialogProps> = ({
+  icon,
+  eventData,
+  onEdit,
+}) => {
+  const [localdata, setLocalData] = useState(eventData);
+  const handleInput = (_: React.ChangeEvent<HTMLInputElement>) =>
+    setLocalData({
+      ...localdata,
+      [_.target.name]: _.target.value,
+    });
+
+  const onDateChange = (date: Date) => {
+    const newDate = date instanceof Date ? date : new Date(date);
```

Comment:
this is where the issue is, it doesn't display the date correctly when old dates are editted to input new dates
```suggestion
    const newDate = date instanceof Date ? date : new Date(date);
```